SpatialGDK/Source/SpatialGDK/Private/Interop/RPCs/CrossServerRPCService.cpp

```diff
@@ -77,73 +78,89 @@ EPushRPCResult CrossServerRPCService::PushCrossServerRPC(Worker_EntityId EntityI
 
 void CrossServerRPCService::AdvanceView()
 {
-	const FSubViewDelta& SubViewDelta = SubView->GetViewDelta();
-	for (const EntityDelta& Delta : SubViewDelta.EntityDeltas)
+	const FSubViewDelta* SubViewDeltas[] = { &ActorSubView->GetViewDelta(), &WorkerEntitySubView->GetViewDelta() };
+	for (auto SubViewDelta : SubViewDeltas)
 	{
-		switch (Delta.Type)
+		for (const EntityDelta& Delta : SubViewDelta->EntityDeltas)
 		{
-		case EntityDelta::UPDATE:
+			AdvanceViewForEntityDelta(Delta);
+		}
+	}
+}
+
+void CrossServerRPCService::AdvanceViewForEntityDelta(const EntityDelta& Delta)
+{
+	switch (Delta.Type)
+	{
+	case EntityDelta::UPDATE:
+	{
+		for (const ComponentChange& Change : Delta.ComponentUpdates)
 		{
-			for (const ComponentChange& Change : Delta.ComponentUpdates)
-			{
-				ComponentUpdate(Delta.EntityId, Change.ComponentId, Change.Update);
-			}
-			break;
+			ComponentUpdate(Delta.EntityId, Change.ComponentId, Change.Update);
 		}
-		case EntityDelta::ADD:
+		break;
+	}
+	case EntityDelta::ADD:
+		PopulateDataStore(Delta.EntityId);
+		break;
+	case EntityDelta::REMOVE:
+	case EntityDelta::TEMPORARILY_REMOVED:
+		CrossServerDataStore.Remove(Delta.EntityId);
+		RPCStore->PendingComponentUpdatesToSend.Remove(
+			EntityComponentId(Delta.EntityId, SpatialConstants::CROSSSERVER_SENDER_ENDPOINT_COMPONENT_ID));
+		RPCStore->PendingComponentUpdatesToSend.Remove(
+			EntityComponentId(Delta.EntityId, SpatialConstants::CROSSSERVER_RECEIVER_ACK_ENDPOINT_COMPONENT_ID));
+		if (Delta.Type == EntityDelta::TEMPORARILY_REMOVED)
+		{
 			PopulateDataStore(Delta.EntityId);
-			break;
-		case EntityDelta::REMOVE:
-		case EntityDelta::TEMPORARILY_REMOVED:
-			CrossServerDataStore.Remove(Delta.EntityId);
-			RPCStore->PendingComponentUpdatesToSend.Remove(
-				EntityComponentId(Delta.EntityId, SpatialConstants::CROSSSERVER_SENDER_ENDPOINT_COMPONENT_ID));
-			RPCStore->PendingComponentUpdatesToSend.Remove(
-				EntityComponentId(Delta.EntityId, SpatialConstants::CROSSSERVER_RECEIVER_ACK_ENDPOINT_COMPONENT_ID));
-			if (Delta.Type == EntityDelta::TEMPORARILY_REMOVED)
-			{
-				PopulateDataStore(Delta.EntityId);
-			}
-			break;
-		default:
-			break;
 		}
+		break;
+	default:
+		break;
```

Comment:
Fail some kind of assertion here? Not sure what the right unreal way is to do that, maybe `ensure(false)`?

Reply:
checkNoEntry()